src/app/[id]/SelectEmails.tsx

```diff
@@ -137,7 +137,16 @@ const SelectEmails = ({ id }: { id: string }) => {
 
         const validatedEmails: Email[] = await Promise.all(
           emails.map(async (email) => {
-            console.log('email', email);
+            const { senderDomain, selector } = await extractEMLDetails(email.decodedContents);
+
+            const response = await fetch('https://archive.zk.email/api/dsp', {
```

Comment:
can you batch/cache these calls to avoid duplication if they're the same? i want to avoid unneeded archive load